hazelcast/src/test/java/com/hazelcast/map/impl/querycache/subscriber/QueryCacheIndexConfigTest.java

```diff
@@ -0,0 +1,47 @@
+package com.hazelcast.map.impl.querycache.subscriber;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.MapIndexConfig;
+import com.hazelcast.config.PredicateConfig;
+import com.hazelcast.config.QueryCacheConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.query.TruePredicate;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class QueryCacheIndexConfigTest extends HazelcastTestSupport {
+
+    @Test
+    public void testIndexConfigIsRespected() {
+        final Config config = new Config();
+        // @formatter:off
```

Comment:
IDE leftover?

Reply:
That's intentional to make the config more easy to read. See Formatter Control section in Preferences -> Editor -> Code Style of IDEA.

Reply:
I think we shouldn't have any IDE-specific directives in our codebase.

Reply:
It's a pretty widespread convention for turning the formatter off, Eclipse also supports this, but it's turned off by default in both IDEA and Eclipse. But you are right, until we have a consensus among all the team member on this feature, one shouldn't use it. Another topic for our future code style settings ;)

Reply:
I'm using the out-of-the-box formatter from IDEA and if I format that config snippet, it still looks like this. So I don't see why this is needed for IDEA.

After 3.10 is out I wanted to have a look at `EditorConfig` or something comparable, so we can define some of these formatting rules in a standard format in the project. For now I think we shouldn't start to add those specific ones shattered in the code.